tests/cpp/test_pointwise.cpp

```diff
@@ -773,4 +783,70 @@ TEST_F(PointwiseTest, VectorizePadLoweringPermuted) {
   EXPECT_TRUE(found_vectorize);
   testValidate(&fusion, cg_outputs, aten_inputs, __LINE__, __FILE__);
 }
+
+TEST_F(PointwiseTest, DomainMapTestEg0) {
+  auto fusion_ptr = std::make_unique<Fusion>();
+  auto fusion = fusion_ptr.get();
+  FusionGuard fg(fusion);
+
+  TensorView* tv0 = makeContigTensor(2);
+  fusion->addInput(tv0);
+  auto tv1 = relu(tv0);
+  fusion->addOutput(tv1);
+  auto tv2 = broadcast(tv1, {false, true, false});
+  auto tv3 = expand(
+      tv2,
+      {tv2->axis(0)->extent(),
+       IrBuilder::create<Val>(4),
+       tv2->axis(2)->extent()});
+  auto tv4 = reshape(tv3, {2, 4, 3}, {2, 12});
+  fusion->addOutput(tv4);
+
+  DomainMapUnitTest domain_map(fusion);
+  // tv1 can't map to tv4
+  EXPECT_FALSE(domain_map.testOutputMapping(tv4, tv1));
+
+  // tv1 can map to tv3
+  EXPECT_TRUE(domain_map.testOutputMapping(tv3, tv1));
+
+  // tv4 can map to tv1
+  EXPECT_TRUE(domain_map.testOutputMapping(tv1, tv4));
+
+  // tv1 is not a valid reference
+  EXPECT_FALSE(domain_map.isValidReference(tv1));
+
+  // tv4 is a valid reference
+  EXPECT_TRUE(domain_map.isValidReference(tv4));
+}
+
+TEST_F(PointwiseTest, DomainMapTestEg1) {
+  auto fusion_ptr = std::make_unique<Fusion>();
+  auto fusion = fusion_ptr.get();
+  FusionGuard fg(fusion);
+
+  TensorView* tv0 = makeContigTensor(2);
+  fusion->addInput(tv0);
+  TensorView* tv1 = makeContigTensor(3);
+  fusion->addInput(tv1);
+  auto tv2 = reshape(tv0, {2, 4}, {8});
+  fusion->addOutput(tv2);
+
+  auto tv3 = broadcast(tv0, {true, false, false});
+  auto tv4 = add(tv1, tv3);
+  fusion->addOutput(tv4);
+
+  DomainMapUnitTest domain_map(fusion);
+  // tv2 can map to tv4, because the missing tv4->axis(0) is a dangling ID.
+  EXPECT_TRUE(domain_map.testOutputMapping(tv4, tv2));
+
+  // tv2 can map to tv4
+  EXPECT_TRUE(domain_map.testOutputMapping(tv2, tv4));
+
+  // However, tv2 is not a valid reference, since it doesn't cover all input IDs
+  EXPECT_FALSE(domain_map.isValidReference(tv2));
```

Comment:
@naoyam This is an example where `tv2` maps to `tv4` can be propagated, so it would pass `areAllOutputIdsMappedTo` check.

But since it doesn't cover all IDs coming from input, it is still rejected as not valid for reference.

Reply:
NVM, Turns out factory method indeed breaks this one. I'll revert that changes.

Reply:
Oh, no 😢